RangeRequest.bs

```diff
@@ -10,26 +10,218 @@ Editor: Chris Lilley, W3C, https://svgees.us/, w3cid 1438
+3. If the client receives the patch-subset magic number, it commences using the patch-subset method. Otherwise, if the client receives the <code>Accept-Ranges: bytes</code> header, it commences using the range-request method. Otherwise, the whole font file is downloaded, and the current non-incremental loading behavior is used.
+
+<h3 id="fallback">
+IFT Method Fallback</h3>
 
 <em>This section is not normative.</em>
 
-Range Request incremental font transfer is an alternate method for incrementally transferring fonts
-that leverages HTTP range requests.
+This summarizes behaviors that result from the above method selection.
+
+<table class="data">
+    <thead>
+        <tr>
+            <th>&nbsp;
+            <th>Client prefers range-request method
+            <th>Client prefers patch-subset method
+    <tbody>
+        <tr><th>Server supports both range-request method and patch-subset method
+            <td>Client makes initial request without query parameter, and possibly with the <code>Range</code> header. Because all patch-subset servers must support the range-request method, the server replies with <code>Accept-Ranges</code> and initial font data. Client/server commence using range-request method.
+            <td>Client makes initial request with query parameter. Server replies with the patch-subset magic number, and client/server commence using patch-subset method.
+        <tr><th>Server supports only range-request method
+            <td>Same as above.
+            <td>Client makes initial request with query parameter. Server replies with <code>Accept-Ranges</code> and initial font data. Client/server commence using range-request method.
+        <tr><th>Server supports neither
+            <td>Client makes initial request without query parameter, and possibly with the <code>Range</code> header. Server replies without <code>Accept-Ranges</code> header, and sends the full font file to the client from beginning to end.
+            <td>Client makes initial request to server with query parameter. Server does not reply with the patch-subset magic number, and sends the full font file to the client from beginning to end.
+</table>
+
+<h2 id="font-organization">
+Font organization</h2>
+
+<h3 id="font-organization-background">
+Background</h2>
+
+<em>This section is not normative.</em>
+
+A particular organization of font files is beneficial for improving the performance of the range-request IFT method. The range-request IFT method only works with [[!TRUETYPE]], [[!OPENTYPE]], [[!WOFF]], or [[!WOFF2]] files. All of these file formats use an <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/otff#organization-of-an-opentype-font"><code>sfnt</code> wrapper</a> which provides a directory of tables inside the font file. A <code>sfnt</code>-based font file is mainly composed of a collection of independent tables.
+
```

Comment:
I get it how range-request can be applied to TTF, OTF and WOFF (on a per-table basis), but I'm not 100% sure I understand how range-request method works with WOFF2 file (that is composed of multiple data streams). This section may need to be extended to include additional details/clarifications.

Reply:
It isn't clear how range request can be applied to WOFF 1 either; it seems to involve compression and decompression to calculate the ranges.

Reply:
IIRC, the WOFF1 file was designed with the range request in mind. It starts with the Header and Table directory where each table is compressed separately, and the table directory provides an offset to the beginning of that table and the compressed length. So, while one cannot ask for a specific glyph range, downloading a set of e.g. metric/layout tables can be separated from the download of the glyph table data. And yes, implementing support for glyph ranges would involve compression and decompression of 'glyf' table.

BTW, on a separate issue, it seems that after the candidate corrections were published for WOFF2 - both [https://www.w3.org/TR/WOFF/](https://www.w3.org/TR/WOFF/) and [https://www.w3.org/TR/WOFF2/](https://www.w3.org/TR/WOFF2/) now point to the same updated WOFF2 Rec.

Reply:
The  https://www.w3.org/TR/WOFF/  to  https://www.w3.org/TR/WOFF2/ redirect was reported and is now fixed

Reply:
The [https://www.w3.org/TR/WOFF/](https://www.w3.org/TR/WOFF/) link still brings me to WOFF2 spec.

Reply:
This appears to be what https://github.com/w3c/IFT/issues/59 is about, so I'll resolve this conversation and add an issue in the PR with a link to that issue.